cleanup-passes/license-header.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import {EOL} from 'os';
+
+import * as promisify from 'promisify-node';
+import * as glob from 'glob';
+
+import {register} from '../cleanup-pass';
+import {ElementRepo} from '../element-repo.ts';
+import {existsSync, makeCommit} from './util';
+
+const polymerHeaderWithoutAtLicense =
+`Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt`;
+
+const polymerHeader =
+`@license
+${polymerHeaderWithoutAtLicense}`;
+
+const fileGlob = '**/*.{css,html,js}';
+
+const shebang = /^#!/;
+const doctype = /^<!\s*doctype/i;
+
+interface GlobOptions {
+  cwd?: string;
+  ignore?: string | string[];
+}
+
+const findFiles: (globPath: string, options?: GlobOptions) => Promise<string[]> =
+    promisify(glob);
+
+function hasLicense(lines: string[], start: number, end: number): boolean {
+  if (start === -1 || end === -1) {
+    return false;
+  }
+  for (let i = start; i < end; i++) {
+    if (lines[i].indexOf('@license') > -1) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function findSpliceLocation(lines: string[], regex: RegExp): number {
```

Comment:
Oops, that should have been `i+1`.
I could use `-1`, and then have a `+1` in `addLicense` and `addLicenseHTML`, since I want to append the license _after_ the doctype.